orchagent/nexthopkey.h

```diff
@@ -98,6 +117,73 @@ struct NextHopKey
     {
         return (ip_address.getV4Addr() == 0);
     }
+
+    bool isMplsNextHop() const
+    {
+        return (label_stack.getSize() > 0);
+    }
+
+    std::string parseMplsNextHop(const std::string& str)
+    {
+        std::string ip_str;
+        auto keys = tokenize(str, LABELSTACK_DELIMITER);
+        if (keys.size() == 1)
+        {
+            // No MPLS info to parse
+            ip_str = str;
+        }
+        else if (keys.size() == 3)
+        {
+            // Expected MPLS format = "<outsegtype>+<labelstack>+<non-mpls-str>"
+            // key[0] = <outsegtype> = "swap" | "pop"
+            // key[1] = <labelstack> = "<label0>/<label1>/../<labelN>"
+            // key[2] = <non-mpls-str> = returned to caller and not parsed here
+            // Example = "push+10100/10101+10.0.0.3@Ethernet4"
+            if (keys[0] == "swap")
```

Comment:
can you point me to code where we add "swap" to the string. I cannot see it .

Reply:
"pop" is a typo... should be "push".  Typo is fixed.

Reply:
@qbdwlr can you please clarify this. Sorry i might be missing while doing walk of your changes.

Reply:
@abdosi "swap" is only available in sonic-swss infra for future expansion.  Currently, since all MPLS routes are triggered from fpmsyncd which has a Linux-centric implementation, there is no true "swap" action.  Linux MPLS assumes implicit pop action on all MPLS routes, so MPLS NHs are always "push"... "swap" in Linux MPLS is inseg pop plus outseg push.  This Linux centric implementation has been confined to fpmsyncd since it is inherently tied to Linux, but orchagent implementation allows for more generic programming in the future.

Reply:
@qbdwlr with new commit i assume above comment does not hold anymore

Reply:
@abdosi Yes, you are correct.  With the new commit, "swap" is now associated with MPLS NHs associated with MPLS routes.  "push" is now associated with MPLS NHs associated with IP routes.